cli/src/template_parser.rs

```diff
@@ -145,6 +148,9 @@ pub enum TemplateParseErrorKind {
     RedefinedFunctionParameter,
     #[error("{0}")]
     Expression(String),
+    // XXX: converted to String so that it becomes Eq
+    #[error("Invalid string pattern: {0}")]
+    StringPattern(String),
```

Comment:
We often attach the original error as the source.
```
TemplateParseError::expression(message, span).with_source(err)
```

Reply:
Yeah, just use `Expression` unless you need special handling of this error type.